temporian/core/serialization.py

```diff
@@ -38,11 +54,90 @@
 
 
 def save(
+    fn: Callable[..., Union[EventSet, Dict[str, EventSet]]],
+    inputs: Dict[str, Any],
+    path: str,
+) -> None:
+    """Saves a Temporian-compiled function to a file.
+
+    Temporian saves the graph built between the function's input and output
+    EventSets or Nodes, not the function itself. Any arbitrary code that is
+    executed in the function will not be ran when loading it back up.
+
+    The function can receive and return other arbitrary parameters, which will
+    be used to trace the function while saving it, but will not be available
+    when loading it back up.
+
+    The function must return an EventSet or Node or a dictionary mapping output
+    names to EventSets or Nodes. If it returns a single one, the output will be
+    saved under the name "output" by default.
+
+    Args:
+        fn: The function to save.
+        inputs: The inputs to pass to the function to trace it, keyed by
+            parameter name. For EventSet/Node parameters, the passed value can
+            be either an EventSet, a Node, or a raw Schema.
+        path: The path to save the function to.
+
+    Raises:
+        ValueError: If the received function is not compiled.
+    """
+    if not fn.is_tp_compiled:
+        raise ValueError(
+            "Can only save a function that has been compiled with"
+            " `@tp.compile`."
+        )
+
+    if not isinstance(inputs, dict):
+        raise ValueError(
+            "`inputs` must be a dictionary mapping the function's parameter"
+            " names to their corresponding values."
+        )
+
+    node_inputs = {}
+    inputs = inputs.copy()
+    for k, v in inputs.items():
+        node_input = _process_fn_input(v)
+        if node_input is not None:
+            inputs[k] = node_input
+            node_inputs[k] = node_input
+
+    outputs = fn(**inputs)

```

Comment:
Should we check exaustively the output type? For example, what is the returned value is a string, or a tuple with an EventSet and a string?